### PR TITLE
misc(compiler): v3 dev-server logging

### PR DESCRIPTION
**Pull request by tanner-reits**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Unit tests (`npm test`) were run locally and passed
- [ ] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [ ] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [X] Other (please describe): Add logging for dev build

## What is the current behavior?

GitHub Issue Number: N/A

## What is the new behavior?

Adds logging statements to the compiler to debug the dev-server reload issue

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: src/compiler/build/watch-build.ts
Comment (rwaskiewicz) [inline src/compiler/build/watch-build.ts:56]: Can we add a logging statement that occurs before we enter this `if` statement (so it always executes?) - I think it could be useful to see cases where `onBuild` is triggered with the `if` statement's conditional evaluating to `true` (which we'd see by the presence of the `console.log`s) and when it evaluating to `false` (which we'd see the proposed console log, but not the ones in the `if` statement)

Comment (rwaskiewicz) [reply src/compiler/build/watch-build.ts:56]: e.g. - https://github.com/ionic-team/stencil/compare/rwaskiewicz/ts-reload-bug-debug#diff-b3739ad1981b6aaadc57ffe972513cf24b1cda208aceccd8a642b0539fc8e878R88

File: src/compiler/build/watch-build.ts
Comment (rwaskiewicz) [inline src/compiler/build/watch-build.ts:116]: Can we also log when this conditional above evaluates to `false`? e.g. https://github.com/ionic-team/stencil/compare/rwaskiewicz/ts-reload-bug-debug#diff-b3739ad1981b6aaadc57ffe972513cf24b1cda208aceccd8a642b0539fc8e878R112

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]
Comparing the branch I had pushed up last week (`rwaskiewicz/ts-reload-bug-debug`), I noticed we're not logging things for `node-sys` - can we port those `console.trace` statements over? https://github.com/ionic-team/stencil/compare/rwaskiewicz/ts-reload-bug-debug#diff-585e96e5364c622963a06a9d71c01a77ad8816bc84e97763dcefe7920176dd50

**Review by rwaskiewicz — COMMENTED** [review]
LGTM. We won't merge this, but I think `@stencil/core@3.1.0-dev.1678134532.41b8f1e` should be good to go if @alicewriteswrongs agrees

**Review by alicewriteswrongs — COMMENTED** [review]
yeah I think this looks good! let's get it going 🚀🐛🚀

**Comment by rwaskiewicz** [comment]
This debug branch PR can be closed - a fix has landed and is included in today's [v2.22.3 release](https://github.com/ionic-team/stencil/releases/tag/v2.22.3) and [v3.2.0 release](https://github.com/ionic-team/stencil/releases/tag/v3.2.0).

**Pull request closed without merging**
